packages/babel-plugin/src/css-prop/__tests__/css-map.test.ts

```diff
@@ -0,0 +1,199 @@
+import type { TransformOptions } from '../../test-utils';
+import { transform as transformCode } from '../../test-utils';
+import { ErrorMessages } from '../../utils/css-map';
+
+describe('css map behaviour', () => {
+  beforeAll(() => {
+    process.env.AUTOPREFIXER = 'off';
+  });
+
+  afterAll(() => {
+    delete process.env.AUTOPREFIXER;
+  });
+
+  const transform = (code: string, opts: TransformOptions = {}) =>
+    transformCode(code, { pretty: false, ...opts });
+
+  const styles = `
+    import { css, cssMap } from '@compiled/react';
+
+    const styles = cssMap({
+      danger: {
+          color: 'red',
+          backgroundColor: 'red'
+      },
+      success: {
+        color: 'green',
+        backgroundColor: 'green'
+      }
+    });
+  `;
+
+  describe('valid syntax', () => {
+    it('should evaluate css map when variant is a runtime variable', () => {
+      const actual = transform(`
+        ${styles}
+
+        <div css={css(styles[variant])} />;
+      `);
+
+      expect(actual).toInclude(
+        '<div className={ax([variant==="danger"&&"_syaz5scu _bfhk5scu",variant==="success"&&"_syazbf54 _bfhkbf54"])}/>'
+      );
+    });
+
+    it('should evaluate css map when variant is statically defined', () => {
+      const actual = transform(`
+       ${styles}
+        
+        <div css={css(styles.success)} />;
+        <div css={css(styles['danger'])} />;
+      `);
+
+      expect(actual).toInclude(
+        '<div className={ax(["success"==="danger"&&"_syaz5scu _bfhk5scu","success"==="success"&&"_syazbf54 _bfhkbf54"])}/>'
```

Comment:
Yea, you are right.
But I think the selected variant is runtime variable at most of cases. Statically access to a variant is "by-product". Users probably should use

```
const a = {
   color: red
}
<div css={css(a)} />
```  

instead of 

```
const styles = {
   a: {
       color: red
   }
}
<div css={css(styles.a)} />
```  
Also, bundler drops the dead code `('apple' === 'orange') && '.aaaabbbb'`